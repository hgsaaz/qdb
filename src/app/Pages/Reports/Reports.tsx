import React from 'react';
import { Tabs, Divider } from 'antd';
import { Helmet, HelmetProvider } from 'react-helmet-async';
import { UserInfo as UserInterFace } from '../PageLayout/types';

const { TabPane } = Tabs;
interface Props {
  user: UserInterFace;
}

const Reports = Props => {
  const { user } = Props;
  return (
    <>
      <HelmetProvider>
        <Helmet titleTemplate="QDB Reports" defaultTitle="QDB Reports">
          <meta name="description" content="QDB application" />
        </Helmet>
      </HelmetProvider>
      <Tabs defaultActiveKey="1">
        <TabPane tab="ACTIVE REPORTS" key="1">
          <h3>Welcome {user && user.name}</h3>
          <Divider />
          <p>You can see Active Reports here</p>
        </TabPane>
        <TabPane tab="CALENDER" key="2">
          <h3>Welcome {user && user.name}</h3>
          <Divider />
          <p>You can see Calender Events here</p>
        </TabPane>
        <TabPane tab="LIVE STATS" key="3">
          <h3>Welcome {user && user.name}</h3>
          <Divider />
          <p>You can see Live Stats here</p>
        </TabPane>
      </Tabs>
    </>
  );
};

export default Reports;
