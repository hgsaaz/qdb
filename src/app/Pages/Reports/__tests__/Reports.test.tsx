import React from 'react';
import { render, act } from '@testing-library/react';
import Reports from '../Reports';

const renderReports = (props: Parameters<typeof Reports>[number] = {}) =>
  render(<Reports {...props} />);
const name = 'testUser';
const testuser = {
  name: name,
};

describe('<Reports />', () => {
  it('should match the snapshot', () => {
    act(() => {
      const { container } = renderReports({
        user: testuser,
      });
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  it('should have props displayed', () => {
    act(() => {
      const { container } = renderReports({
        user: testuser,
      });
      expect(container.querySelector('h3')).toMatchSnapshot();
    });
  });
});
