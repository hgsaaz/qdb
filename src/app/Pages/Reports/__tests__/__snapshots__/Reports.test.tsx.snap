// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Reports /> should have props displayed 1`] = `
<h3>
  Welcome 
  testUser
</h3>
`;

exports[`<Reports /> should match the snapshot 1`] = `
<div
  class="ant-tabs ant-tabs-top"
>
  <div
    class="ant-tabs-nav"
    role="tablist"
  >
    <div
      class="ant-tabs-nav-wrap"
    >
      <div
        class="ant-tabs-nav-list"
        style="transform: translate(0px, 0px);"
      >
        <div
          class="ant-tabs-tab ant-tabs-tab-active"
        >
          <div
            aria-selected="true"
            class="ant-tabs-tab-btn"
            role="tab"
            tabindex="0"
          >
            ACTIVE REPORTS
          </div>
        </div>
        <div
          class="ant-tabs-tab"
        >
          <div
            aria-selected="false"
            class="ant-tabs-tab-btn"
            role="tab"
            tabindex="0"
          >
            CALENDER
          </div>
        </div>
        <div
          class="ant-tabs-tab"
        >
          <div
            aria-selected="false"
            class="ant-tabs-tab-btn"
            role="tab"
            tabindex="0"
          >
            LIVE STATS
          </div>
        </div>
        <div
          class="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
        />
      </div>
    </div>
    <div
      class="ant-tabs-nav-operations ant-tabs-nav-operations-hidden"
    >
      <button
        aria-controls="null-more-popup"
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-hidden="true"
        class="ant-tabs-nav-more"
        id="null-more"
        style="visibility: hidden; order: 1;"
        tabindex="-1"
        type="button"
      >
        <span
          aria-label="ellipsis"
          class="anticon anticon-ellipsis"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="ellipsis"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
  <div
    class="ant-tabs-content-holder"
  >
    <div
      class="ant-tabs-content ant-tabs-content-top"
    >
      <div
        aria-hidden="false"
        class="ant-tabs-tabpane ant-tabs-tabpane-active"
        role="tabpanel"
        tabindex="0"
      >
        <h3>
          Welcome 
          testUser
        </h3>
        <div
          class="ant-divider ant-divider-horizontal"
          role="separator"
        />
        <p>
          You can see Active Reports here
        </p>
      </div>
      <div
        aria-hidden="true"
        class="ant-tabs-tabpane"
        role="tabpanel"
        style="display: none;"
        tabindex="-1"
      />
      <div
        aria-hidden="true"
        class="ant-tabs-tabpane"
        role="tabpanel"
        style="display: none;"
        tabindex="-1"
      />
    </div>
  </div>
</div>
`;
